packages/flutter/test/material/time_picker_test.dart

```diff
@@ -222,6 +222,48 @@ void main() {
     expect(selectedLabels.map<bool>((dynamic tp) => tp.inner as bool), inner0To23);
   });
 
+  // Regression test for https://github.com/flutter/flutter/issues/164860
+  testWidgets('Material3 - formats 24-hour numbers correctly in Farsi', (
+    WidgetTester tester,
+  ) async {
+    await mediaQueryBoilerplate(
+      tester,
+      locale: const Locale('fa', 'IR'),
+      materialType: MaterialType.material3,
+    );
+
+    final MaterialLocalizations localization = MaterialLocalizations.of(
+      tester.element(find.byType(TimePickerDialog)),
+    );
+
+    final List<String> labels00To23 = List<String>.generate(24, (int index) {
```

Comment:
Can we instead use the literal strings for the numbers for comparison? This way it's clearer what we're looking for, and avoids the current way which feels like just repeating the implementation code.

Reply:
Sure, pushing required change.